### DISCUSSION TITLE
Add option to choose different OSC settings for full screen and windowed

### DISCUSSION THREAD

**Discussion by advice2020** [mpv-player/mpv#14699] [Q&A] [answered]
Add the option to allow user the ability to choose a different OSC setting for when using FULL SCREEN and for when WINDOWED.

Examples for two options I would be interested in seeing have separate OSC setting for would be...
visibility
boxvideo

`FULLSCREEN`
visibility=auto
I would like to be able to set OSC to only show up when mouse movement is detected.
boxvideo=no
I would like OSC to overlay on top of video

However would like to be able to have different settings when MPV is WINDOWED

`WINDOWED`
visibility=always
I would like to be able to set OSC always show.
boxvideo=yes
I would like OSC be below video (instead of overlaid on top of video)

I was wondering if any developers would consider adding these options.

Thank You

**Comment by guidocella** [comment, marked as answer]
https://mpv.io/manual/master/#conditional-auto-profiles

**Comment by advice2020** [comment]
Thank you for your response
I looked at this but this seems over my head, I do not think I have the ability to do this.
Also correct me if I am wrong but when I here profile I am thinking that maybe this applies only when loading MPV.
I apologize if I did not mentioned but I am looking for the ability for example to start off in FULLSCREEN using OSC and then be able to change to WINDOWED mode in the same opened instance and then have the OSC switch to being underneath the media and be showing all the time.
If you or anyone else have any other ideas, or if anything I mention above is not correct, please let me know.
Thank you again

**Reply by guidocella** [reply]
That's what conditional profiles are for.

**Comment by advice2020** [comment]
@guidocella
Thank you for your response.
Ahh I apologize when I read about this elsewhere it made it sound like this required Lua knowledge.
Also when I hear profiles I think of it as having to restart the program to load a different profile.
I did some guessing and I was able to get this to work.
Thank you again for pointing out this feature.

`To Anyone Interested In The Topic`
Only thing is when you solve a problem sometimes it leads to another one, was so happy to figure out how to do this only to then notice the following.
Just realized that "boxvideo" adds black bars to left and right of videos which is not desired, there is an old issue report about this requesting a fix but does not appear to be being worked on.
Keep this in mind before using/trying out the following as it contains "boxvideo"

Here is an example of how to do this if anyone is interested in this in the future.
Can add the following to your mpv.conf file
```
[fullscreen]
profile-cond=fullscreen
script-opts=osc-visibility=auto,osc-boxvideo=no

[window]
profile-cond=not fullscreen
script-opts=osc-visibility=always,osc-boxvideo=yes
```
Hopefully this will be helpful to someone in the future.
